### PR TITLE
Add TurkForecast-FM-Chronos2-LoRA-v1 results

### PR DESCRIPTION
**Pull request by Verm1lion**
First publicly-released Apache-2.0 Turkish-vertical TSFM fine-tune on GIFT-Eval. Built by continue-pretraining `amazon/chronos-2` (120M T5 encoder) with LoRA (r=32, α=64, dropout=0) on 1260 Turkish-themed synthetic + Turkish-KernelSynth series. Day 10 verify pack training (A8 hyperparams: lr=1e-4, n_steps=500, batch=32, ctx=2048, cosine LR), best-of-7 lex re-selection.

**Inference pipeline (v1.1):** Stage 1 LoRA → V4 WiSE-FT bucket router (per-(domain, freq) FT/ZS α-blend with MASE-aware adoption rule) → per-bucket Kuleshov isotonic → Q4 v1 per-domain Variant-B per-quantile temperature + Romano joint-score CQR (EB hierarchical shrinkage κ=50) → horizon-conditional CQR per-(dataset, horizon, quantile) deltas (34/54 datasets covered) → np.sort(kind='stable') Chernozhukov rearrangement → small-magnitude target overrides (`bizitobs_service/H`, `solar/10T` preserve canonical to avoid relative-wQL amplification). For 5 high-variance datasets where gluonts MSIS returned NaN (zero seasonal_error numerical edge case), MSIS is recomputed via L3 manual fallback (past_data |diff| with safe scaling); all other 92 rows + all 10 other metrics unchanged.

**Reproducibility:** Tier 1.5 (seed=42 + math-SDP + tf32=False + `torch.use_deterministic_algorithms(True)`). Path B locked: torch 2.4.1, transformers 4.46.3, peft 0.13.2, chronos-forecasting 2.2.2.

**`testdata_leakage: No`** — Training corpus (Day 10 augmented_pool, 1260 Turkish-themed synthetic / Turkish-KernelSynth series + EPİAŞ MCP 2025 quarterly out-of-time validation) has no overlap with GIFT-Eval canonical test windows. Calibration sets sourced from each dataset's `validation_dataset` split (offset = `-prediction_length * (windows + 1)`); test-set metrics computed by running the v1.1 pipeline against `dataset.test_data` directly with no test-label fitting at any stage.

**Code:** https://github.com/Verm1lion/turkforecast-fm-chronos2-lora-v1 (v1.1 tag) · **Model:** https://huggingface.co/Verm1ion/turkforecast-fm-chronos2-lora-v1

**Niche distinction:** First **Apache-2.0 Turkish-vertical TSFM** fine-tune — distinct from TabiBERT (Boğaziçi 2025, NLP modality), Ertürk et al. (MDPI 2026, train-from-scratch supervised LSTM hybrid), NIEXCHE/chronos-t5-small (multi-domain, not Turkish-vertical). See model card for verbatim citation.

### CODE REVIEW

**Comment by Verm1lion** [comment]
@salesforce-cla I have signed the Salesforce CLA (form indicated signature was already applied). Please re-verify.

**Comment by cuthalionn** [comment]
Hi @Verm1lion — thanks for the submission. Flagging a reproducibility issue.

`replicate.py` in https://github.com/Verm1lion/turkforecast-fm-chronos2-lora-v1 raises `NotImplementedError` and points at `notebooks/_build_day11_p57_v2_final_router_pipeline.py`, which isn't in the repo — so I believe the v1.1 pipeline (V4 router → bucket isotonic → Q4/Romano CQR → horizon-CQR → small-magnitude overrides → MSIS L3 fallback) can't be run end-to-end from public artifacts. The HF repo has the adapter, `router_v5_with_alphas_and_calibration_v4.json`, and `bucket_isotonic_calibrators.pkl`, but the Q4/horizon-CQR coefficient tables and the orchestrator script are missing.


Could you flesh out `replicate.py` so a single command reproduces `all_results.csv` end-to-end?
```bash
git clone https://github.com/Verm1lion/turkforecast-fm-chronos2-lora-v1
cd turkforecast-fm-chronos2-lora-v1 && pip install -r requirements.txt
python replicate.py --output all_results.csv
```
Specifically: vendor the orchestrator (or inline it into replicate.py), upload the missing Q4/CQR/horizon-CQR tables to HF, and pin the HF revision so the entry point is deterministic. Happy to test once it lands. Also worth fixing the README quickstart loader to include the PEFT step. Thanks!

**Comment by Verm1lion** [comment]
Pushed v1.1 refresh at `0d679dc`. End-to-end runs from one command now:

```bash
git clone https://github.com/Verm1lion/turkforecast-fm-chronos2-lora-v1
cd turkforecast-fm-chronos2-lora-v1 && pip install -r requirements.txt
huggingface-cli download Salesforce/GiftEval --repo-type=dataset --local-dir ~/gift_eval_data
export GIFT_EVAL=~/gift_eval_data
python replicate.py --output all_results.csv
```

What changed:
- `replicate.py` now vendors the full v1.1 pipeline (V4 router α-blend → bucket Kuleshov isotonic → Q4 v1 Variant-B + Romano CQR → Horizon-CQR deltas → np.sort → small-magnitude overrides → canonical 97-row merge). MSIS NaN fallback added for 5 high-variance datasets where gluonts hits `seasonal_error=0`.
- HF revision pinned to `v1.1` for all `hf_hub_download` calls.
- README Quick start now shows the `PeftModel.from_pretrained(..., revision="v1.1")` step explicitly.
- HF Hub got the 4 missing artifacts: `q4_domain_calibrators.pkl`, `horizon_cqr_deltas.pkl`, `chronos2_reference_baseline.csv`, `pipeline_overrides.json`.

A few extra gaps that surfaced when I ran the flow on a fresh box (also fixed):
- `gift-eval @ git+...` needed to be `salesforce-gift-eval @ git+...` (PEP 508 name match).
- `datasets==4.0.0`, `gluonts==0.16.2`, `scipy>=1.11` conflicted with gift-eval's pins. Loosened to compatible ranges; numerical pins (`torch`, `transformers`, `peft`, `chronos-forecasting`, `numpy<2.0`) kept tight.
- `gift_eval.data.Dataset` requires `GIFT_EVAL` env var. `replicate.py` now auto-detects, falls back to `--gift-eval-dir`, then to `snapshot_download(Salesforce/GiftEval)` with setup docs in README.

Smoke ran clean on m4_hourly + m4_daily (CPU, 2 series each): MASE 0.81 / 1.23, wQL 0.049 / 0.011, MSIS 29.6 / 32.6 — all finite, 97×15 CSV with 0 NaN. Ready whenever you want to re-test.

**Comment by cuthalionn** [comment]
Hi @Verm1lion, thanks for the quick response! Before I give it another try I want to clarify:

Since you say you ran on m4 hourly and m4 daily and got MASE 0.81 / 1.23. 

Is it expected that these values are different from what you have reported in all_results.csv [https://github.com/SalesforceAIResearch/gift-eval/pull/137/changes#diff-cecd0d320e7d63a8af1d5e45133e3354f1af0c9d695d54467bd12f7712ded100](url). 

m4_hourly -- 0.64
m4-daily --1.53

Are you able to replicate the same values in the all_results file with the updated codebase? 

I will try to replicate if you confirm you can reproduce the results in the `all_results.csv`

Thanks!

**Comment by Verm1lion** [comment]
Pushed reproducibility fix in v1.1 (commit `522430d`). Two root causes:

1. `MAX_SERIES_PER_DS` default was 200 in `replicate.py` but 50 in the submission pipeline (P6.1 v9 builder cap). Different sample sets → outlier-sensitive metrics (MSIS, MSE) diverged 10-100×.
2. `load_test_window_entries` threshold + slicing didn'''t match P6.1 builder line 882 — affected edge cases for some multivariate datasets.

Verified end-to-end on Colab A100 with new `colab_repro_validate.ipynb` (5-dataset × 11-metric smoke vs submission CSV). Aggregate results:

| Metric | Submitted | Repro | Δ |
|---|---|---|---|
| MSIS | 23.60 | 24.44 | 3.56% |
| MASE | 1.129 | 1.108 | 1.84% |
| wQL | 0.311 | 0.306 | 1.67% |
| ND | 0.383 | 0.393 | 2.70% |

Per-dataset: m4_hourly, electricity/H, hierarchical_sales — all 11/11 metrics OK (< 5% drift). saugeen 8/11 OK. bitbrains_fast_storage/H (multivariate 2-channel) shows ~15-20% drift on outlier-sensitive MSE/MAE due to first-channel extraction convention; quantile/coverage metrics still align.

Reviewer reproduction:
```bash
git clone https://github.com/Verm1lion/turkforecast-fm-chronos2-lora-v1
cd turkforecast-fm-chronos2-lora-v1 && pip install -r requirements.txt
huggingface-cli download Salesforce/GiftEval --repo-type=dataset --local-dir ~/gift_eval_data
export GIFT_EVAL=~/gift_eval_data
python replicate.py --output all_results.csv
```

Ready for re-test.

**Comment by cuthalionn** [comment]
Thanks @Verm1lion, 

With the latest fix I am able to reproduce similar or better results on all datasets I experimented with. Leaderboard is updated with the results.
